File: 1411-convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
int getDecimalValue(struct ListNode* head) {
    int res=0;
    int n=0;
    struct ListNode *temp=head;
    while(temp!=NULL){
        n++;
        temp=temp->next;
    }
    while(head!=NULL){
        res+=(head->val)*pow(2,n-1);
        n--;
        head=head->next;
    }
    return res;
}

File: 1411-convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def getDecimalValue(self, head):
        temp=head;
        n=0;
        while(temp!=None):
            n+=1;
            temp=temp.next;
        o=0;
        while(head!=None):
            o+=head.val*(2**(n-1));
            n-=1;
            head=head.next;
        return o;

        """
        :type head: Optional[ListNode]
        :rtype: int
        """
        